docs/gen-ai/openai.md

```diff
@@ -0,0 +1,162 @@
+| [`gen_ai.openai.request.response_format`](/docs/attributes-registry/gen-ai.md) | string | The response format that is requested. | `json` | `Conditionally Required` if the request includes a response_format | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.openai.request.seed`](/docs/attributes-registry/gen-ai.md) | int | Requests with same seed value more likely to return same result. | `100` | `Conditionally Required` if the request includes a seed | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.openai.request.service_tier`](/docs/attributes-registry/gen-ai.md) | string | The service tier requested. May be a specific tier, detault, or auto. | `auto`; `default`; `standard` | `Conditionally Required` if the request includes a service_tier | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.openai.response.service_tier`](/docs/attributes-registry/gen-ai.md) | string | The service tier used for the response. | `standard` | `Conditionally Required` if the request includes a service_tier | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`server.port`](/docs/attributes-registry/server.md) | int | GenAI server port. [6] | `80`; `8080`; `443` | `Conditionally Required` If `server.address` is set. | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`gen_ai.request.frequency_penalty`](/docs/attributes-registry/gen-ai.md) | double | The frequency penalty setting for the GenAI request. | `0.1` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.request.max_tokens`](/docs/attributes-registry/gen-ai.md) | int | The maximum number of tokens the model generates for a request. | `100` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.request.presence_penalty`](/docs/attributes-registry/gen-ai.md) | double | The presence penalty setting for the GenAI request. | `0.1` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.request.stop_sequences`](/docs/attributes-registry/gen-ai.md) | string[] | List of sequences that the model will use to stop generating further tokens. | `["forest", "lived"]` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.request.temperature`](/docs/attributes-registry/gen-ai.md) | double | The temperature setting for the GenAI request. | `0.0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.request.top_k`](/docs/attributes-registry/gen-ai.md) | double | The top_k sampling setting for the GenAI request. | `1.0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
I don't think openai has this parameter - foes it?

Reply:
It is included because I extended the genai span which has it. Is there a way for mt override its inclusion, or do I need to remove the extends and copy the attribute list?

Reply:
There is an example in https://github.com/open-telemetry/semantic-conventions/pull/1393 - effectively we add a new group with attributes used everywhere and then extend this group in system-specific conventions. 

I wonder if top_k is in fact a generic attribute though - it seems it's not applicable to a bunch of systems - but that's out of scope of this PR.

Reply:
wait, it seems `top_logprobs` is the OpenAI parameter that's aligns with Anthropic and Cohere top_k

[OpenAI top_logprobs](https://platform.openai.com/docs/api-reference/chat/create#chat-create-top_logprobs):

> An integer between 0 and 20 specifying the number of most likely tokens to return at each token position, each with an associated log probability. logprobs must be set to true if this parameter is used.


[Cohere k](https://docs.cohere.com/reference/chat#request.body.k): 
> Ensures only the top k most likely tokens are considered for generation at each step. Defaults to 0, min value of 0, max value of 500.

[Anthropic top_k](https://docs.anthropic.com/en/api/messages#:~:text=top_k,integer):

> Only sample from the top K options for each subsequent token.
> Used to remove "long tail" low probability responses. [Learn more technical details here](https://towardsdatascience.com/how-to-sample-from-language-models-682bceb97277).

Reply:
The closest parameter in OpenAI to the other two is *top_p*. Both seek to limit the number of likely tokens used in each prediction. *top_k* sorts by the probability and simply takes the top k number of tokens, then choses a next token.  *top_p* uses a distribution and then cuts off the tokens when CDF exceeds p. However, in these cases you still get one set of token completions. 

*top_logprobs* has a different behavior. This parameter combined with the boolean *logprobs* parameter, will return n number of tokens for each prediction, along with their probabilities for the client to do some processing with.  Each prediction has multiple tokens. 

I did add *top_p* to the openai set in my last set of commits. However, I think we may want to tackle *logprobs* in its own PR because there could also be design considerations on how the response is captured.